drone/src/agent/docker.rs

```diff
@@ -231,12 +231,12 @@ impl DockerInterface {
         })
     }
 
-    #[allow(unused)]
     pub async fn pull_image(
         &self,
         image: &str,
         credentials: &Option<DockerCredentials>,
     ) -> Result<()> {
+        let timer = Timer::new();
```

Comment:
I really like the idea of this, is there any way to implement it without the extra noise? would an attribute macro work?

Reply:
I wonder if any timing info is built into tracing's built-in [attribute macro](https://docs.rs/tracing-attributes/latest/tracing_attributes/attr.instrument.html). Eventually we should probably instrument the code with that as it's an ergonomic way to add spans that work across await calls.